webapp/tests/karma/ts/services/telemetry.service.spec.ts

```diff
@@ -0,0 +1,265 @@
+    storageGetItemStub = sinon.stub(window.localStorage, 'getItem');
+    storageSetItemStub = sinon.stub(window.localStorage, 'setItem');
+    storageRemoveItemStub = sinon.stub(window.localStorage, 'removeItem');
+
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: DbService, useValue: dbService },
+        { provide: SessionService, useValue: sessionService }
+      ]
+    });
+
+    service = TestBed.inject(TelemetryService);
+    dbService = TestBed.inject(DbService);
+    sessionService = TestBed.inject(SessionService);
+    clock = sinon.useFakeTimers(NOW);
+    window.PouchDB = () => pouchDb;
+  });
+
+  afterEach(() => {
+    clock.restore();
+    sinon.restore();
+  });
+
+  describe('record()', () => {
+    it('should record a piece of telemetry', async () => {
+      pouchDb.post = sinon.stub().resolves();
+      pouchDb.close = sinon.stub();
+      storageGetItemStub
+        .withArgs('medic-greg-telemetry-db')
+        .returns('dbname');
+      storageGetItemStub
+        .withArgs('medic-greg-telemetry-date')
+        .returns(Date.now().toString());
+      
+      await service.record('test', 100);
+
+      expect(consoleErrorSpy.callCount).to.equal(0);
+      expect(pouchDb.post.callCount).to.equal(1);
+      expect(pouchDb.post.args[0][0]).to.deep.include({ key: 'test', value: 100 });
+      expect(pouchDb.post.args[0][0].date_recorded).to.be.above(0);
+      expect(storageGetItemStub.callCount).to.equal(2);
+      expect(pouchDb.close.callCount).to.equal(1);
+    });
+
+    it('should default the value to 1 if not passed', async () => {
+      pouchDb.post = sinon.stub().resolves();
+      pouchDb.close = sinon.stub();
+      storageGetItemStub
+        .withArgs('medic-greg-telemetry-db')
+        .returns('dbname');
+      storageGetItemStub
+        .withArgs('medic-greg-telemetry-date')
+        .returns(Date.now().toString());
+
+      await service.record('test');
+
+      expect(consoleErrorSpy.callCount).to.equal(0);
+      expect(pouchDb.post.args[0][0].value).to.equal(1);
+      expect(pouchDb.close.callCount).to.equal(1);
+    });
+
+    it('should set localStorage values', async () => {
+      pouchDb.post = sinon.stub().resolves();
+      pouchDb.close = sinon.stub();
+      storageGetItemStub
+        .withArgs('medic-greg-telemetry-db')
+        .returns(undefined);
+      storageGetItemStub
+        .withArgs('medic-greg-telemetry-date')
+        .returns(undefined);
+
+      await service.record('test', 1);
+
+      expect(consoleErrorSpy.callCount).to.equal(0);
+      expect(storageSetItemStub.callCount).to.equal(2);
+      expect(storageSetItemStub.args[0][0]).to.equal('medic-greg-telemetry-db');
+      expect(storageSetItemStub.args[0][1]).to.match(/medic-user-greg-telemetry-/); // ends with a UUID
+      expect(storageSetItemStub.args[1][0]).to.equal('medic-greg-telemetry-date');
+      expect(storageSetItemStub.args[1][1]).to.equal(NOW.toString());
```

Comment:
How come this changed?

Reply:
@dianabarsan What piece in specific changed? I'm a bit confused. Thanks!

Reply:
The assertion in the original test was:
```
expect(storageSetItemStub.args[1][1]).to.equal(NOW);
```
https://github.com/medic/cht-core/blob/master/webapp/tests/karma/unit/services/telemetry.js#L125

Reply:
Ah it's because you're explicitly converting to string in the migrated code, vs the old code: https://github.com/medic/cht-core/blob/master/webapp/src/js/services/telemetry.js#L66

Reply:
Aaah yes, because Typescript complains that [localStorage setItem ](https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem) only accepts string as value. So I had to do a parsing :)